packages/commonwealth/server/util/roles.ts

```diff
@@ -188,109 +206,55 @@ export async function findOneRole(
   return role;
 }
 
-export async function createDefaultCommunityRoles(
-  models: DB,
-  chain_id: string
-): Promise<void> {
-  // Create default roles
-  try {
-    await models.CommunityRole.create({
-      chain_id,
-      name: 'member',
-      allow: BigInt(0),
-      deny: BigInt(0),
-    });
-    await models.CommunityRole.create({
-      chain_id,
-      name: 'moderator',
-      allow: BigInt(0),
-      deny: BigInt(0),
-    });
-    await models.CommunityRole.create({
-      chain_id,
-      name: 'admin',
-      allow: BigInt(0),
-      deny: BigInt(0),
-    });
-  } catch (error) {
-    throw new Error(`Couldn't create default community roles ${error}`);
-  }
-}
-
 export async function createRole(
   models: DB,
   address_id: number,
   chain_id: string,
-  role_name?: Permission,
+  role_name?: Role,
   is_user_default?: boolean,
   transaction?: Transaction
 ): Promise<RoleInstanceWithPermission> {
-  if (role_name === undefined) {
-    role_name = 'member';
-  }
-
-  // check if role is already assigned to address
-  const communityRoles = await findAllCommunityRolesWithRoleAssignments(
-    models,
-    { where: { address_id } },
-    chain_id
-  );
-  if (communityRoles.findIndex((r) => r.name === role_name) !== -1) {
-    // if role is already assigned to address, return current highest role this address has on that chain
-    const highestCommunityRole: CommunityRoleAttributes =
-      await getHighestRoleFromCommunityRoles(communityRoles);
-    if (
-      highestCommunityRole.RoleAssignments &&
-      highestCommunityRole.RoleAssignments.length > 0
-    ) {
-      const roleAssignment = highestCommunityRole.RoleAssignments[0];
-      const role = new RoleInstanceWithPermission(
-        roleAssignment,
-        chain_id,
-        highestCommunityRole.name,
-        highestCommunityRole.allow,
-        highestCommunityRole.deny
-      );
-      return role;
-    } else {
-      throw new Error('No role found');
-    }
-  }
-
-  // Get the community role that has given chain_id and name if exists
-  let community_role;
-  community_role = await models.CommunityRole.findOne({
-    where: { chain_id, name: role_name },
+  is_user_default = !!is_user_default;
+  if (!role_name) return; // all users are default member, so they cannot be changed
```

Comment:
Basically old code had:
```
  if (role_name === undefined) {
    role_name = 'member';
  }
 ```
 
At the top of the function. But this did nothing because member was the lowest role, and this method updates the role to be the highest between existing role and specified role. So it was better to just return early.